Add gateway architecture page to docs (#3008)
hyperledger/fabric: docs/source/discovery-overview.rst

the needed information dynamically and present it to the SDK in a consumable
manner.

> Starting with v2.4, the :doc:`gateway` interacts with service discovery in order to determine

Reviewer: "Starting with Fabric v2.4, the :doc:`Fabric Gateway` ... etc."

> which peers are required to endorse a transaction, and which orderer nodes to send

Reviewer: ```suggestion
which peers are required to endorse a transaction, and which ordering nodes to send
```

Reviewer: both are used in the literature. ordering nodes is 3x more common than orderer.

> the transaction.  Consequently, client applications written using the new gateway

Reviewer: "... to send the transaction to."

Reviewer: "...written using the new Fabric Gateway ... "
(i.e. let's be formal about it at first because it's a new thing. Over time it can revert to more informal "gateway")

> the transaction to.  Consequently, client applications written using the new gateway

Reviewer: ```suggestion
the transaction to. Client applications written using the Fabric Gateway
```

> SDKs do not need to interact with service discovery at all.

Reviewer: "do not need to" implies that they still can. Can they still interact with discovery? 
... alternatives are "do not interact" or "should not interact" ...

Author: changed to "do not interact directly with..."

> SDKs do not interact directly with service discovery at all.

Reviewer: ```suggestion
SDKs do not interact directly with service discovery.
```
